### ISSUE TITLE
Iteration Plan for September 2019

### ISSUE THREAD

**Issue by nrajpurkar** [microsoft/appcenter#970]
# September 2019 Plan

Android developers and Dark Mode lovers rejoice! We spent last month finishing out our feature set for Android App Bundle support with Xamarin.Android build support and automatic store distribution after a build. Dark Mode is officially in preview to shield your eyes from the bright, white light while perfecting your apps. It was also a great month for those of you with WPF and WinForms apps with added support for Diagnostics and Analytics.

We’re continuing work on some of the features we started last month including bringing your own identity management provider to Auth, support for self-hosted GitLab instances, and in-app updates through public distribution groups. We’ll also be tackling support for building apps with Xamarin.iOS app extensions. Your testers can look forward to a beautiful, updated experience when interacting with the app releases distributed to them. We’re looking to ease moving apps in bulk from HockeyApp to App Center as well as enabling some app management pieces directly in App Center. Lastly, we’re exploring new features such as a free tier in Data and Compute Backend-as-a-Service.

### Auth
- [x] Support bringing your own identity management provider (Firebase Authentication and Auth0) with App Center Auth [#637]
- [x] Enable developers to connect their Active Azure Directory to Auth [#994]

### Build
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11 beta 6
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11 beta 7
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11 GM
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11 GM 2
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11 GA
- [x] Support for building Swift/Obj-C and React Native iOS apps with Xcode 11.1 GM
- [x] Support for building Xamarin apps with Mono version: 6.4.0.198, Xamarin.iOS: 13.2.0.42, Xamarin.Android: 10.0.0.43, Xamarin.Mac:  6.2.0.42 [#1096 ]
- [x] Support for Xamarin.iOS App Extensions [#30]
- [x] Self-Hosted GitLab Support [#37]

### Data
- [x] Data Metrics Explorer [#660]
- [ ] Investigatory work on Free Tier for App Center Data [#799]
- [x] Data Explorer [#800]

### Diagnostics
- [ ] API to upload custom crashes [#31]
- [ ] Support macOS & iOS extensions [#687]
- [ ] Crashes and errors support for UWP apps [#752, #150, #403]
- [ ] User ID support for WPF, WinForms, and UWP apps

### Distribution
- [ ] ~Support Sparkle feeds for macOS/Windows in-app updates [#202]~
- [ ] ~In-app update through public distribution groups [#182]~
- [ ] ~Updated testere experience [#986, #140, #369, #176]~
- [x] New CodePush SDK [#814]

### SDK
- [x] Support Carthage in the App Center Apple SDKs [#977]
- [ ] ~Add diagnostics (crashes) support for iOS extensions [#687]~

### HockeyApp
- [x] Switch the owner management of organizations from HockeyApp to App Center
- [x] Allow to move multiple apps simultaneously via API

### General
- [ ] Research for compute backend-as-a-service from App Center [#993]

### Test
- [ ] ~Add iOS 13 devices to the device selection [#1129]~

**Comment by sgreifeneder** [comment]
As mentioned by @patniko in #792, please consider #30 to be part of the September iteration plan!

**Comment by Zee-day** [comment]
Can you please bring issue #32 into this iteration?

**Comment by nrajpurkar** [comment]
Working on it this month @sgreifeneder. Which unfortunately limits our capacity to work on #32 @Zee-day. Definitely still on our radar.

**Comment by fabiendem** [comment]
Hey, it looks like "In-app update through public distribution groups [#182]" was in the August roadmap but couldn't get completed. https://github.com/microsoft/appcenter/issues/792

It's not part of the September roadmap at the moment.
Could you just clarify what to expect from such items not completed yet and dropped from the future roadmaps?

Thanks!

**Comment by nrajpurkar** [comment]
Hey @fabiendem, the issue has been pulled into this month's iteration plan. We haven't finished finalizing this yet.

**Comment by masatoru** [comment]
What does it mean that the task of your own identity management provider is complete? When can I use this feature?

**Comment by masatoru** [comment]
I received an email replying to this question. Thanks.
https://docs.microsoft.com/en-us/appcenter/distribution/groups#adding-azure-active-directory-aad-groups-to-a-distribution-group

**Comment by patniko** [comment]
@masatoru Firebase Authentication, Auth0 and AAD support are on their way!

**Comment by beeradmoore** [comment]
@nrajpurkar ,  Xcode 11 went GM (and then GM2) a week ago. Is it safe to presume the Xcode 11 beta 6 and Xcode 11 beta 7 tasks above are being done against Xcode 11 GM 2?

**Comment by nrajpurkar** [comment]
Good catch @beeradmoore, we're actually already on GM 2. I fixed that above.

**Comment by beeradmoore** [comment]
Awesome. Thanks :)

**Comment by pkhandelwal-cdl** [comment]
I checked the GitHub logs, is there a timetable to support building Xamarin.iOS with XCode 11?

**Comment by nrajpurkar** [comment]
@pkhandelwal-cdl we're working on support for the latest Xamarin.iOS version at the moment and hope to ship that soon. I'll add it to our iteration plan.

**Comment by follesoe** [comment]
Awesome to see that you keep this information open on GitHub. Found the answer I was looking for (you are working on Xcode 11 GM + latest Xamarin.iOS support in the current iteration). Keep up the good work with the AppCenter platform! 👍

**Comment by MaximePost** [comment]
> Hey, it looks like "In-app update through public distribution groups [#182]" was in the August roadmap but couldn't get completed. #792
> 
> It's not part of the September roadmap at the moment.
> Could you just clarify what to expect from such items not completed yet and dropped from the future roadmaps?
> 
> Thanks!

Critically waiting on that basic functionality to actually work

**Comment by pkhandelwal-cdl** [comment]
Looks like support for building Xamarin.iOS for XCode 11 is now live! Thank you AppCenter team for the quick turnaround!!

**Comment by lamathe** [comment]
@nrajpurkar Do you have an ETA for the issue under Diagnostics > API to upload custom crashes ?

**Comment by winnieli1** [comment]
Hi @lamathe - we should have the endpoint available in the next few weeks!

**Comment by DrBeak1** [comment]
I don't see an October plan -- are these still being posted regularly @nrajpurkar ?

**Comment by nrajpurkar** [comment]
@DrBeak1 yep, October plan is over at #1114

**Comment by DrBeak1** [comment]
Thanks @nrajpurkar! I was looking [here](https://github.com/microsoft/appcenter/wiki/Iteration-Plans) and didn't see it.

**Comment by mertnesvat** [comment]
Is there any update regarding https://github.com/microsoft/appcenter/issues/615 it's closed but no idea how to fix, checked many issues and asked to customer support but no reply, does any body have an idea?
